libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/RustMatrixClient.kt

```diff
@@ -486,20 +486,26 @@ class RustMatrixClient(
         verificationService.destroy()
 
         sessionDelegate.clearCurrentClient()
-        innerRoomListService.destroy()
-        notificationService.destroy()
+        innerRoomListService.close()
+        notificationService.close()
         notificationProcessSetup.destroy()
-        encryptionService.destroy()
-        innerClient.destroy()
+        encryptionService.close()
+        innerClient.close()
     }
 
     override suspend fun getCacheSize(): Long {
         return baseDirectory.getCacheSize()
     }
 
-    override suspend fun clearCache() {
-        close()
-        deleteSessionDirectory(deleteCryptoDb = false)
+    override suspend fun clearCache(isEventCacheEnabled: Boolean) {
+        if (isEventCacheEnabled) {
```

Comment:
Are you sure we need to check for event cache? iOS is not

Reply:
I'd just remove our manual clear cache and use only the `clear_cache` from sdk.

Reply:
OK, done in https://github.com/element-hq/element-x-android/pull/4396/commits/c857a3b6f48a67fbda8060137d84661db957ce88

Reply:
> [clear_caches()](https://github.com/matrix-org/matrix-rust-sdk/blob/5a22944f5298a622c0f8d9b2a64d254d6312a07d/bindings/matrix-sdk-ffi/src/client.rs#L1192) is doing a better job than our "manual" clear cache, that just deletes everything. I guess we can keep it like this and when the flag about the event cache will be removed, just keep the call to `clear_caches`. Do you agree?

Actually, I think this is what was done on iOS and suggested by the SDK team (sorry for the late feedback, I just saw the discussion).